server/events/policy_filter.go

```diff
@@ -11,31 +11,83 @@ type prReviewsFetcher interface {
 	ListApprovalReviewers(ctx context.Context, installationToken int64, repo models.Repo, prNum int) ([]string, error)
 }
 
+type teamMemberFetcher interface {
+	ListTeamMembers(ctx context.Context, installationToken int64, teamSlug string) ([]string, error)
+}
+
 type ApprovedPolicyFilter struct {
-	//TODO: remove in favor of an in-memory cache of owners
-	GlobalCfg        valid.GlobalCfg
-	PRReviewsFetcher prReviewsFetcher
+	owners map[string][]string //cache
```

Comment:
Do you want to use sync.Map for this?  Currently this would be problematic when used across multiple go routines.